akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java

```diff
@@ -1200,4 +1200,32 @@ public void mustRunSourceAndIgnoreElementsItOutputsAndOnlySignalTheCompletionWit
             .join();
     assertEquals(Done.getInstance(), completion);
   }
+
+  @Test
+  public void flattenOptional() {
+    // #flattenOptional
+    final List<Integer> resultList =
+        Source.range(1, 10)
+            .map(x -> Optional.of(x).filter(n -> n % 2 == 0))
+            .via(Flow.flattenOptional())
+            .runWith(Sink.seq(), system)
+            .toCompletableFuture()
+            .join();
```

Comment:
I would prefer a timeout on such blocking operations, instead of `join`.
```
public void flattenOptional() throws Exception {
  ...
  get(5, TimeUnit.SECONDS);
```

I see that some other tests here already use `join`, and maybe you can change them at the same time.

and we shouldn't show blocking code in documentation, so end the   `// #flattenOptional` after `runWith`

Reply:
I have updated, thanks for the detailed review.